File: src/main/java/com/example/the_machine/dto/ContentBlockTypes.java
package com.example.the_machine.dto;

/**
 * Constants for ContentBlock type identifiers. These constants ensure consistency between the
 * type() method return values and the @JsonSubTypes.Type name attributes.
 */
public final class ContentBlockTypes {

  public static final String TEXT = "text";
  public static final String MARKDOWN = "markdown";
  public static final String TOOL_CALL = "tool_call";
  public static final String TOOL_RESULT = "tool_result";
  public static final String IMAGE_REF = "image_ref";
  public static final String FILE_REF = "file_ref";

  private ContentBlockTypes() {
    // Prevent instantiation
  }
}

File: src/main/kotlin/com/example/the_machine/dto/content/ContentBlockTypes.kt
package com.example.the_machine.dto.content

/**
 * Constants for ContentBlock type identifiers. These constants ensure consistency between the
 * type() method return values and the @JsonSubTypes.Type name attributes.
 */
object ContentBlockTypes {

  const val TEXT = "text"
  const val MARKDOWN = "markdown"
  const val TOOL_CALL = "tool_call"
  const val TOOL_RESULT = "tool_result"
  const val IMAGE_REF = "image_ref"
  const val FILE_REF = "file_ref"
}